### PR TITLE
pl - added parallelization

### PR DESCRIPTION
**Pull request by peterlee6925**
Added parallelization for KDE/ISDE in PLoM_libary.py

### CODE REVIEW

**Comment by sanjayg0** [comment]
@peterlee6925 , Peter ,
Can you summarize the changes that you have made?  That way it will be easier to understand the changes that you are proposing.  Comments in the code are of course also very helpful, too :)  Additionally have you tested your changes against the examples provided in the code base to ensure that the results are correct, and can you say anything about the speed ups you are seeing by using Pool.

Also I notice that part of the pull request includes a file PLoM_library.py.bak.  Did you intend that to be the case?

@kuanshi , Kuanshi can you have a look also at Peter's proposed changes.

**Comment by kuanshi** [comment]
Hi Prof. Govindjee,

Sorry about the confusion - Peter is the PhD student that I mentioned previously working with Prof. Deierlein on the PEER project on surrogate modeling of bridge earthquake response.   He is recently working on the parallelism of two computationally heavy steps (KDE and ISDE) and made really good improvements for the computational efficiency for really high-dimension and large sample size problems (50,000+ samples), so I suggested to make a pull request to make a contribution to the main repo.

And thank you so much for providing the prompt review and suggestions - yes, I'll work with Peter to (1) double check the results of all examples to make sure they are consistent, (2) clean up codes and add more comments, (3) report the runtime case studies to discuss the improvement, and (4) add more details to the pull request comment.

Thanks again for your kind review and suggestions! We'll update our progress to you and please feel free to let us know if you have any other comments or suggestions!

With gratitude,
Kuanshi


________________________________

**Comment by peterlee6925** [comment]
Hi Prof. Govindjee,

Thank you for the prompt reply, and all the feedback! Kuanshi and I are meeting tomorrow, and will add more details/verify the changes. 

Best,

Peter
